smolyay/basis.py

```diff
@@ -1,29 +1,33 @@
 import abc
-import math
 
 import numpy
 import scipy.special
 
+
 class BasisFunction(abc.ABC):
     """Basis function for interpolating data.
 
-     A one-dimensional basis function is defined on the domain
-     :math:`[-1,1]`. The function defines the :attr:`points` at
-     which it should be sampled within this interval for interpolation.
-     The function also has an associated :meth:`__call__` method
-     for evaluating it at a point within its domain. Moreover,
-     the first derivative of the function can be evaluated via
-     :meth:`derivative`.
+    A one-dimensional basis function is defined on some given natural 
+    domain. The function defines the :attr:`points` at which it should 
+    be sampled within this interval for interpolation. The function also 
+    has an associated :meth:`__call__` method for evaluating it at a 
+    point within its domain. Moreover, the first derivative of the function 
+    can be evaluated via :meth:`derivative`.
+
+    Parameters
+    ----------
+    natural_domain : list of two numbers
+        the natural domain of the function within which calls are made.
     """
 
-    def __init__(self):
+    def __init__(self, natural_domain):
```

Comment:
Two things:
1. I think this would be better called `domain`.
2. I would make this an abstract property that has to be overridden by deriving classes like:
```py
@property
@abc.abstractmethod
def domain(self):
    pass
```